### ISSUE TITLE
Cloud interface learning curve

### ISSUE THREAD

**Issue by marminatix** [blynkkk/beta#18]
Hi.
First of all I've to tell you I'm not a skilled software programmer but I oftenly use Blynk in a basic manner to demonstrate to managers my work results as operation's consultant and the quantitative impact of my methods on company's productivity.

The "only-android" suite is very fast to understand and to learn to use, very friendly and really effective in terms of work/result ratio. A lot of people were astonished looking at the easyness with which I reached the goals of measuring the performances on the workfloor via Blynk and a Arduino with some sensors.

Now, I know the future is on the cloud and multiplatform but, as a rookie, I've found some relevant issues (and spent some hours) in finding data and activating procedures (for example, simply finding the auth token); and I knew the procedures from the app system! Can't imagine the impact on new users with my background, my skillness and my targets to reach. 

If you want to keep this fantastic simplicity of the app-approach, my humble suggestion is to improve the "first impact" of the cloud system to non experienced users. Maybe guiding new users the first time they use a menu or a tool, adapting some kind of live guideline to the target the need to achieve could be useful. 
People like me approaches these tools without so much time to spend in clicking from menu to menu and trying to understand the difference between a template and a device. We need a flexible, scalable and "pret a porter" tool that give "simple" answers with a right quantity of time investment.

Thank you very (very!) much for your great ideas.
Marco

**Comment by doom369** [comment]
@marminatix thanks for the feedback. I agree, we already reworked some UI screens few times and this is never-ending ongoing work. Every sprint we implement dozens of improvements and we have an onboarding guide in our plans. Still, a lot of work to do.

**Comment by Peterkn2001** [comment]
I tend to agree with some of @marminatix's comments and have some thoughts to share, so I'll use this issue as the place...

**Templates**
I find the Template/Device approach slows development down quite a bit if you're creating a one-off device, or projects that use substantially different devices.
It's obviously going to be a quicker way to work if you're attempting to deploy dozens of identical devices, but that's not where this group of testers is coming from.
It's a bit like having to create a template every time you want to create a new word document!

**Datastreams**
I like the idea of being able to give meaningful names to datastreams, but the process of swapping backwards and forwards between the app and the web dashboard to add a new datastream then push it out to the devices before being able to add it to a widget to be quite cumbersome. It slows-down `quick and dirty` approach - which might not be a bad thing, but it's a big change from what we're used to.
The ability to add a datastream from the app would make life simpler, with less switching between devices.

**Web Dashboard**
I think this is a real disappointment. At the very least I'd expected it to allow a web view of the existing app widgets automatically, preferably with the ability to edit those widgets from the web browser.
Clearly it's going to serve as a high level monitor/control platform eventually, but for the 'home' user it's going to have very little value.

**App Layout**
The approach of forcing the use of Tiles isn't going to suit everyone. It would be better if the tiles were more configurable, so that you could add multiple high-level data items that you might need - gauges, multiple LED to show statuses, display widgets etc.
In addition, the app grid still has a lot of shortfalls as far as I'm concerned. Although widgets are more positionable, the minimum sizes of some widgets (LEDs in particular) is still too large for my liking.

**Beta Testing Approach**
The approach of "here you go guys, get on with it", combined with the "there is some documentation, but it wont help much" isn't going to create the level of engagement that you're hoping for from many people.
I've never run a beta testing programme that didn't have comprehensive Getting Started guides, plus some sort of testing script or checklist to show which areas of functionality have been tested. Your current approach of letting/expecting testers to do their own thing will obviously create a diverse approach, as everyone will have their own pet things that they want to look at - assuming that they get past the initial hurdles.
However, you only know about the issues that people are encountering. There is system for testers to record the areas that they've tested and received satisfactory results. This means that at the end of the process, you wont know which areas have actually be tested, and which haven't.

**Beta Testing Timelines**
Pavel mentioned a launch date of "I think the end of April".
There's no published timetable for the first batch of beta testers to complete their testing, but you've said that "We just invited the first 100 users. After getting some first feedback we’ll invite another 1000 users and so on." I don't know if the One Thousand users was a typo, but if not then that's going to get interesting!
Realistically, from what I've seen so far, the product isn't going to be available to launch to the marketplace by the end of April. Even if the product was totally free, launching it that soon would damage the future of the product/company in my opinion.

And finally, by biggest single complaint - the icons for device tiles includes this....
![image](https://user-images.githubusercontent.com/35718564/112765095-79efea80-9003-11eb-96ca-e61f72329da3.png)
Seriously, what type of people hang their toilet paper this way around!

Pete.
